docs/source/examples/example_dates.rst

```diff
@@ -0,0 +1,158 @@
+Example: Dates
+==============
+
+This example concerns itself with expressing ``Constraint`` s against data revolving
+around dates. While date ``Constraint`` s between tables exist, we will only illustrate
+``Constraint`` s on a single table and reference values here. As a consequence, we will
+only use ``WithinRequirement``, as opposed to ``BetweenRequirement``.
+
+Concretely, we will assume a table containing prices for a given product.
+Importantly, these prices are valid for a certain date range only. More precisely,
+we assume that the price is indicated in the ``price`` column, the date from which
+it is valid - the date itself included - in ``date_from`` and the the until when
+it is valid - the date itself included - in the ``date_to`` column.
+
+Such a table might look as follows:
+
+.. list-table:: prices
+   :header-rows: 1
+
+   * - price
+     - date_from
+     - date_to
+   * - 13.99
+     - 22/01/01
+     - 22/01/10
+   * - 14.5
+     - 22/01/11
+     - 22/01/17
+   * - 13.37
+     - 22/01/16
+     - 22/01/31
+
+Given this table, we would like to ensure that 6 constraints are satisfied:
+
+1. All values from column ``date_from`` should be in January 2022.
+2. All values from column ``date_to`` should be in January 2022.
+3. The minimum value in column ``date_from`` should be the first of January 2022.
+4. The maximum value in column ``date_to`` should be the 31st of January 2022.
```

Comment:
Maybe you want to explain why you want to have those? Particularly, 1(2) and 3(4) are basically the same. Is there any advantage to doing it like that or just for educational purposes to extract more constraints to show?

Reply:
> or just for educational purposes to extract more constraints to show

Exactly! Just as the other examples, these are merely illustrations of datajudge functionalities - as opposed to real life case studies.

Reply:
Done